plugins/backup/backroll/src/main/java/org/apache/cloudstack/backup/BackrollBackupProvider.java

```diff
@@ -0,0 +1,390 @@
+                backupToUpdate.setStatus(Backup.Status.BackedUp);
+                if(backupDao.persist(backupToUpdate) != null) {
+                    s_logger.info("Backroll mise à jour enregistrée");
+                    backupDao.remove(backupToFind.getId());
+                }
+            }
+        }
+
+        // delete deleted backroll backup in CS
+        backupsInDb = backupDao.listByVmId(null, vm.getId());
+        for (Backup backup : backupsInDb) {
+            String backupName = backup.getExternalId().contains(",") ? backup.getExternalId().split(",")[1] : backup.getExternalId();
+            BackrollVmBackup backupToFind = backupsFromBackroll.stream()
+                .filter(backupInBackroll -> backupInBackroll.getName().contains(backupName))
+                .findAny()
+                .orElse(null);
+
+            if(backupToFind == null) {
+                BackupVO backupToUpdate = ((BackupVO) backup);
+                backupToUpdate.setStatus(Backup.Status.Removed);
+                if(backupDao.persist(backupToUpdate) != null) {
+                    s_logger.debug("Backroll suppression enregistrée (sync)");
+                }
+            }
+        }
+    }
+
+    @Override
+    public String getConfigComponentName() {
+        return BackupService.class.getSimpleName();
+    }
+
+    @Override
+    public ConfigKey<?>[] getConfigKeys() {
+        return new ConfigKey[]{
+            BackrollUrlConfigKey,
+            BackrollAppNameConfigKey,
+            BackrollPasswordConfigKey
+        };
+    }
+
+    @Override
+    public boolean deleteBackup(Backup backup, boolean forced) {
+        s_logger.info("backroll delete backup id: " + backup.getExternalId());
+        if(backup.getStatus().equals(Backup.Status.BackingUp)) {
+            throw new CloudRuntimeException("You can't delete a backup while it still BackingUp");
+        } else {
+            s_logger.debug("backroll - try delete backup");
+            VMInstanceVO vm = vmInstanceDao.findByIdIncludingRemoved(backup.getVmId());
+
+            if(backup.getStatus().equals(Backup.Status.Removed) || backup.getStatus().equals(Backup.Status.Failed)){
+                return deleteBackupInDb(backup);
+            } else {
+                if (getClient(backup.getZoneId()).deleteBackup(vm.getUuid(), getBackupName(backup))) {
+                    s_logger.debug("backroll delete backup ok on backroll side");
+                    return deleteBackupInDb(backup);
+                }
+            }
+        }
+        return false;
+    }
+
+    private boolean deleteBackupInDb(Backup backup) {
+        BackupVO backupToUpdate = ((BackupVO) backup);
+        backupToUpdate.setStatus(Backup.Status.Removed);
+        if(backupDao.persist(backupToUpdate) != null) {
+            s_logger.debug("Backroll backup delete in database");
+            return true;
+        }
+        return false;
+    }
+
+    protected BackrollClient getClient(final Long zoneId) {
```

Comment:
Looking at the implementation, we will only be able to have a single backroll client for all ACS zones, is that correct?